analyzer/codechecker_analyzer/analyzer.py

```diff
@@ -140,8 +139,7 @@ def perform_analysis(args, skip_handlers, actions, metadata_tool,
     analyzers = args.analyzers if 'analyzers' in args \
         else analyzer_types.supported_analyzers
     analyzers, errored = analyzer_types.check_supported_analyzers(analyzers)
-    analyzer_types.check_available_analyzers(analyzers, errored)
-
+    analyzer_types.check_available_analyzers(analyzers, errored, args)
```

Comment:
I had a rather hard project a while ago, trying to factor `args` out of a function mostly responsible for `CodeChecker cmd diff` (#3863). The problem with `args` is that, in my mind, its supposed to be a very temporary object -- the 'main' file or function receives this object as a representation of the user's program arguments, parses it, and discards it.

How would you unit test a function that requires monstrositty of an object? You'd spend 30 minutes trying to assemble all the small little parts of args until the test finally runs. If you decide to thread `args` out of a function, and all the functions that it calls that also take `args`, you'll spend about a week doing it.

I am perfectly aware that I'm speaking of greater evils than what is seen here (pretty much a slippery slope argument), but these, imo, decent reasons against it :^)